src/security/Guardian.sol

```diff
@@ -0,0 +1,179 @@
+     *      This function pauses repay, withdraw, borrow, deposit and liquidation.
+     *      This function can only be called by the guardian.
+     *      Guardian can only pause the protocol once 32 days past from the last pause. This is to prevent
+     *  guardian from pausing the protocol too often. And giving unpause time for users to trigger.
+     *  When protocol is paused for 30 days only owner role has a right to unpause the protocol. After 30 days,
+     *  any user can unpause the protocol. This flow gives at least 2 days to unpause the protocol for any user
+     *  since guardian can only trigger pause once every 32 days after the previous pause event.
+     */
+    function pause() external onlyGuardian {
+        require(block.timestamp > pauseTimestamp + 32 days, "Guardian: Cannot pause, Pause time not expired");
+        repayPaused = true;
+        withdrawPaused = true;
+        borrowPaused = true;
+        depositPaused = true;
+        liquidationPaused = true;
+        pauseTimestamp = block.timestamp;
+        emit PauseUpdate(msg.sender, repayPaused, withdrawPaused, borrowPaused, depositPaused, liquidationPaused);
+    }
+
+    /**
+     * @param repayPaused_ Whether repay functionality should be paused.
+     * @param withdrawPaused_ Whether withdraw functionality should be paused.
+     * @param borrowPaused_ Whether borrow functionality should be paused.
+     * @param depositPaused_ Whether deposit functionality should be paused.
+     * @dev Unpauses repay, withdraw, borrow, and deposit functionality.
+     *      This function can unPause variables individually.
+     *      Only owner can call this function. It updates the variables if incoming variable is false.
+     *  If variable is false and incoming variable is true, then it does not update the variable.
+     */
+    function unPause(
+        bool repayPaused_,
+        bool withdrawPaused_,
+        bool borrowPaused_,
+        bool depositPaused_,
+        bool liquidationPaused_
+    ) external onlyOwner {
+        repayPaused = repayPaused && repayPaused_;
+        withdrawPaused = withdrawPaused && withdrawPaused_;
+        borrowPaused = borrowPaused && borrowPaused_;
+        depositPaused = depositPaused && depositPaused_;
+        liquidationPaused = liquidationPaused && liquidationPaused_;
+        emit PauseUpdate(msg.sender, repayPaused, withdrawPaused, borrowPaused, depositPaused, liquidationPaused);
+    }
+
+    /**
+     * @dev Users can call this function after 30 days that the protocol is paused. Since Guardian can only pause the protocol
+     *  once every 32 days, this function gives at least 2 days to any user to unpause the protocol.
+     *      This function can unPause variables all at once.
+     */
+    function unPause() external {
+        require(block.timestamp > pauseTimestamp + 30 days, "Guardian: Cannot unPause, unPause time not expired");
```

Comment:
```suggestion
        require(block.timestamp > pauseTimestamp + 30 days && block.timestamp < pauseTimestamp + 32 days, "G_UP: Cannot unPause");
```

Reply:
I think we shouldn't limit users to 30-32 days. If owner still not paused after 32nd day, Users should still be able to unpause it.